clients/unshaded/src/main/java/tachyon/client/block/BufferedBlockInStream.java

```diff
@@ -29,13 +30,13 @@
 
 /**
  * Provides a stream API to read a block from Tachyon. An instance extending this class can be
- * obtained by calling {@link TachyonBlockStore#getInStream(long)}. The buffer size of the stream
- * can be set through configuration. Multiple {@link BufferedBlockInStream}s can be opened for a
- * block. This class is not thread safe and should only be used by one thread.
+ * obtained by calling {@link TachyonBlockStore#getInStream}. The buffer size of the stream can be
+ * set through configuration. Multiple {@link BufferedBlockInStream}s can be opened for a block.
  *
  * This class provides the same methods as a Java {@link InputStream} with additional methods from
  * Tachyon Stream interfaces.
  */
+@NotThreadSafe
```

Comment:
I think we should be explicit. I created a JIRA https://tachyon.atlassian.net/browse/TACHYON-1542 to add these annotations to the rest of our code base.

There are several problems with using implicit annotations:
1) not everyone reading the code base will know what the default is
2) automated tools suggest FindBugs use these annotations (e.g. inconsistent synchronization check is not performed for classes annotated with `@NotThreadSafe`
3) when people forget to add the annotation, the "implicit" annotation might be incorrect ... making thread safety annotations explicit forces developers to think about thread safety, which is a good thing